src/kakarot/instructions/system_operations.cairo

```diff
@@ -548,6 +597,101 @@ namespace CallHelper {
 }
 
 namespace CreateHelper {
+    // @notice Constructs an evm contract address for the create2 opcode
+    //         via last twenty bytes of the keccak hash of:
+    //         keccak256(0xff + sender_address + salt +
+    //         keccak256(initialisation_code))[12:]
+    //         See [CREATE2](https://www.evm.codes/#f5)
+    // @param sender_address - The evm sender address.
+    // @param bytecode_len - The length of the initialization code.
+    // @param bytecode - The offset to store the element at.
+    // @param salt - The salt given to the create2 opcode.
+    // @return The pointer to the updated calling context.
+    func get_create2_address{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr,
+        bitwise_ptr: BitwiseBuiltin*,
+    }(sender_address: felt, bytecode_len: felt, bytecode: felt*, salt: Uint256) -> (
+        evm_contract_address: felt
+    ) {
+        alloc_locals;
+        let (keccak_ptr: felt*) = alloc();
+        local keccak_ptr_start: felt* = keccak_ptr;
+
+        let (local bytecode_bytes8: felt*) = alloc();
+        Helpers.bytes_to_bytes8_little_endian(
+            bytes_len=bytecode_len,
+            bytes=bytecode,
+            index=0,
+            size=bytecode_len,
+            bytes8=0,
+            bytes8_shift=0,
+            dest=bytecode_bytes8,
+            dest_index=0,
+        );
+        with keccak_ptr {
+            // get keccak hash of bytecode
+            let (bytecode_hash_bigend) = keccak_bigend(
+                inputs=bytecode_bytes8, n_bytes=bytecode_len
+            );
+            // get keccak hash of
+            // marker + caller_address + salt + bytecode_hash
+            let (local concat_args_bytes: felt*) = alloc();
+            // 0xff is by convention the marker involved in deterministic address creation for create2
+
+            let (concat_args_bytes_len) = Helpers.felt_to_bytes(0xff, 0, concat_args_bytes);
+
+            let (concat_args_bytes_len) = Helpers.felt_to_bytes(
+                sender_address, concat_args_bytes_len, concat_args_bytes + concat_args_bytes_len
+            );
+
+            let (salt_bytes_array_len, salt_bytes_array) = Helpers.uint256_to_bytes_array(salt);
+            memcpy(
+                dst=concat_args_bytes + concat_args_bytes_len,
+                src=salt_bytes_array,
+                len=salt_bytes_array_len,
+            );
+
+            let (
+                bytecode_hash_bytes_array_len, bytecode_hash_bytes_array
+            ) = Helpers.uint256_to_bytes_array(bytecode_hash_bigend);
+            memcpy(
```

Comment:
I had the thought of writing a helper variant of `Helpers.uint256_to_bytes_array` that writes the bytes array to a destination array.

It might help make this code more readable, as otherwise you have to 'manually' track the length of `concat_args_bytes`

Reply:
yes, I agree it feels like we could have another helper that would factorize this